redisless/src/server/tests/mod.rs

```diff
@@ -77,21 +77,31 @@ fn test_redis_implementation() {
 
 #[test]
 #[serial]
-fn expire() {
+fn expire_and_ttl() {
     let port = 3359;
     let server = Server::new(InMemoryStorage::new(), port);
     assert_eq!(server.start(), Some(ServerState::Started));
     let redis_client = redis::Client::open(format!("redis://127.0.0.1:{}/", port)).unwrap();
     let mut con = redis_client.get_connection().unwrap();
 
+    let ttl: i32 = con.ttl("key").unwrap();
+    assert_eq!(ttl, -2);
+    let ttl: i32 = con.pttl("key").unwrap();
+    assert_eq!(ttl, -2);
     // EXPIRE
-    let duration: usize = 2;
+    let duration: usize = 50;
     let _: () = con.set("key", "value").unwrap();
     let x: String = con.get("key").unwrap();
     assert_eq!(x, "value");
+    let ttl: i32 = con.ttl("key").unwrap();
+    assert_eq!(ttl, -1);
 
     let ret_val: u32 = con.pexpire("key", duration).unwrap();
     assert_eq!(ret_val, 1);
+    let ttl: i32 = con.pttl("key").unwrap();
+    assert!(ttl <= duration as i32 && duration as i32 - 3 < ttl);
+    let ttl: i32 = con.ttl("key").unwrap();
+    assert_eq!(ttl, (duration / 1000) as i32);
```

Comment:
This could potentially fail to assert in a case in which you go from 4000 to 3999, for example, where now you get left = 3 and right = 4, even though that behavior is fine.

Reply:
This is correct. Any idea how to improve it but setting a non round duration?